code/framework/utils/src/main/java/io/cattle/platform/util/exception/LoggableException.java

```diff
@@ -0,0 +1,7 @@
+package io.cattle.platform.util.exception;
+
+import org.slf4j.Logger;
+
+public interface LoggableException {
+    void log(Logger log);
```

Comment:
I think you can just have a `log()` message and the exception is responsible for logging itself with it's own logger.  Being that we use this in one place in NoExceptionRunnable, it doesn't seem like the logger we pass in is going to greatly enhance the information in the log